tensorboard/data/provider.py

```diff
@@ -248,6 +248,42 @@ def read_scalars(
         """
         pass
 
+    @abc.abstractmethod
+    def read_last_scalars(
+        self,
+        ctx=None,
+        *,
+        experiment_id,
+        plugin_name,
```

Comment:
Since we're requiring plugin_name to be specified, I would think having the method be read_latest_values, or something like this, would be more generic/flexible, and we can say in the doc string that it's possible that not all plugins are supported by all implementations.

Although the return value does get a bit more difficult to specify and deal with, so if you prefer this because of this reason, I'm ok with it.

Reply:
I specified `plugin_name` because `read_scalars` also requires `plugin_name`. I prefer the simpler definition for just the scalars, in this case do you think we can just remove the `plugin_name` arg?

Reply:
I decided to remove the `plugin_name` completely. I think it's unlikely that we'll have a new plugin type that contains scalar data in the near future. And since scalar seems to be the most used type, we don't have to generalize it to `read_last_values` just yet, wdyt?

Reply:
Oh, sorry for the late response. I thought I had replied to this. Technically, the plugin type is somewhat independent from the data type, e.g. there's a `custom_scalars` plugin in our examples, I believe, which also uses scalar values.

For this reason, and to be consistent with the rest of the interface for the data provider, I'd lean a bit towards keeping the plugin_name... but this is ok too. I agree it's unlikely we'll ever need anything other than "scalars", and if we do, we can change it later (but again, unlikely).

Reply:
Great point, added it back.

Reply:
Also I didn't realize I also needed to implement this method in `event_processing/data_provider`. PTAL, thanks!